unity-renderer/Assets/Scripts/MainScripts/DCL/WorldRuntime/SceneResourcesLoadTracker/Tests/MockedSharedComponent.cs

```diff
@@ -0,0 +1,105 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using DCL.Components;
+using DCL.Controllers;
+using DCL.Models;
+
+namespace Tests
+{
+    public class MockedSharedComponent : ISharedComponent
```

Comment:
Cannot this be mocked using NSubstitute?

Reply:
I don't think I can.
I need to mock that the shared component is ready,
there is no event for that in the interface
just a method for subscription `CallWhenReady`.
You are the NSubstitute master, any ideas? I would also love to remove this class :)